Source/Views/PageView.swift

```diff
@@ -170,19 +169,28 @@ class PageView: UIScrollView {
 
     let imageViewSize = imageView.frame.size
     let imageSize = image.size
-    let realImageViewSize: CGSize
 
+    // Size the imageView so that it has the same aspect ratio as the image, while using the
+    // maximum possible width or height.
+    let newImageViewSize: CGSize
     if imageSize.width / imageSize.height > imageViewSize.width / imageViewSize.height {
-      realImageViewSize = CGSize(
+      newImageViewSize = CGSize(
         width: imageViewSize.width,
         height: imageViewSize.width / imageSize.width * imageSize.height)
     } else {
-      realImageViewSize = CGSize(
+      newImageViewSize = CGSize(
         width: imageViewSize.height / imageSize.height * imageSize.width,
         height: imageViewSize.height)
     }
 
-    imageView.frame = CGRect(origin: CGPoint.zero, size: realImageViewSize)
+    imageView.frame = CGRect(origin: CGPoint.zero, size: newImageViewSize)
+
+    // We zoom the image out if necessary to leave space for the header / footer to be visible.
+    let maxHeight = imageViewSize.height - 120
+    if (maxHeight > 0 && imageView.frame.height > maxHeight) {
+      minimumZoomScale = maxHeight / imageView.frame.height
+      zoomScale = minimumZoomScale
```

Comment:
Why are we "zooming out" instead of changing the bounds to account for the height? Semantically, I think the header/footer are essentially just some top/bottom padding in the default state.